### PR TITLE
Private registry tests pipeline should support ephemeral Harbor instance

### PR DESCRIPTION
**Pull request by pshirodkar-orcl**
# Description

Adds support for ephemeral Harbor instance. Removes support for standalone harbor registry.

Fixes VZ-4272

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [x] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -10,7 +10,7 @@ def VERRAZZANO_DEV_VERSION = ""
 def tarfilePrefix=""
 def storeLocation=""
 
-def agentLabel = env.JOB_NAME.contains('master') ? "phxlarge" : "VM.Standard2.8"
+def agentLabel = "largeexperimental"
```
Comment (theabmitra) [inline Jenkinsfile:13]: General comment: This needs to be reverted once all tests are successful

Comment (pshirodkar-orcl) [reply Jenkinsfile:13]: Yes waiting for current set of builds to pass. Once the builds are complete, I will proceed to revert the similar changes from all affected jenkins files.

File: ci/scripts/prepare_private_registry_jenkins_at_environment.sh [hunk 2]
```diff
@@ -1,101 +1,195 @@
...
-    echo "Installation Failed trying to apply the Verrazzano CR YAML"
+deploy_contour () {
+  kubectl apply -f https://projectcontour.io/quickstart/contour.yaml
+  kubectl patch daemonsets -n ${PROJECTCONTOUR_NAMESPACE} envoy -p '{"spec":{"template":{"spec":{"nodeSelector":{"ingress-ready":"true"},"tolerations":[{"key":"node-role.kubernetes.io/master","operator":"Equal","effect":"NoSchedule"}]}}}}'
+}
+
+install_new_helm_version() {
+  cd ${WORKSPACE}
+  # Downloading newer version of helm in order to prevent harbor installation issues
+  helm_zip="helm-v3.7.2-linux-amd64.tar.gz"
```
Comment (theabmitra) [inline ci/scripts/prepare_private_registry_jenkins_at_environment.sh:29]: convert helm version into a global variable, so we can reuse this for later releases

File: ci/scripts/prepare_private_registry_jenkins_at_environment.sh [hunk 3]
```diff
@@ -1,101 +1,195 @@
...
+  echo "Downloading helm 3.7.2 via command: curl -fsSL -o ${helm_zip} https://get.helm.sh/${helm_zip}"
```
Comment (theabmitra) [inline ci/scripts/prepare_private_registry_jenkins_at_environment.sh:30]: use the variable to reference the version

File: ci/scripts/prepare_private_registry_jenkins_at_environment.sh [hunk 4]
```diff
@@ -1,101 +1,195 @@
...
+  curl -fsSL -o ${helm_zip} https://get.helm.sh/${helm_zip}
+  if [ $? -ne 0 ]; then
+    echo "Error downloading helm 3.7.2"
```
Comment (theabmitra) [inline ci/scripts/prepare_private_registry_jenkins_at_environment.sh:33]: use variable to reference helm version

File: ci/scripts/prepare_private_registry_jenkins_at_environment.sh [hunk 5]
```diff
@@ -1,101 +1,195 @@
...
+    exit 1
+  fi
+}
+
+deploy_harbor() {
+  cd ${WORKSPACE}/linux-amd64
+  # Install harbor via new version of helm
+  ./helm --kubeconfig=${KUBECONFIG} repo add harbor https://helm.goharbor.io
+  ./helm --kubeconfig=${KUBECONFIG} repo update
+  ./helm --kubeconfig=${KUBECONFIG} install ephemeral-harbor harbor/harbor \
```
Comment (theabmitra) [inline ci/scripts/prepare_private_registry_jenkins_at_environment.sh:57]: convert harbor release into a global variable
HARBOR_RELEASE=ephemeral-harbor
helm install  ${HARBOR_RELEASE} ....

File: ci/scripts/prepare_private_registry_jenkins_at_environment.sh [hunk 6]
```diff
@@ -1,101 +1,195 @@
...
+    --set expose.ingress.hosts.core=${REGISTRY} \
+    --set expose.ingress.annotations.'kubernetes\.io/ingress\.class'=contour \
+    --set externalURL=http://${REGISTRY} \
+    --set expose.tls.secretName=ephemeral-harbor-ingress-cert \
+    --set notary.enabled=false \
```
Comment (theabmitra) [inline ci/scripts/prepare_private_registry_jenkins_at_environment.sh:62]: Remove space

File: ci/scripts/prepare_private_registry_jenkins_at_environment.sh [hunk 7]
```diff
@@ -1,101 +1,195 @@
...
+    --set harborAdminPassword=${PRIVATE_REGISTRY_PSW}
+
+  sleep 10
+
+  kubectl wait --namespace default \
+  --for=condition=ready pod \
+  --all \
+  --timeout=300s
+
+  echo "waiting for contour to show up"
```
Comment (theabmitra) [inline ci/scripts/prepare_private_registry_jenkins_at_environment.sh:75]: "Wait for contour pods to be up and running"

File: ci/scripts/prepare_private_registry_jenkins_at_environment.sh [hunk 8]
```diff
@@ -1,101 +1,195 @@
...
+  ./tests/e2e/config/scripts/create-image-pull-secret.sh ocr "${OCR_REPO}" "${OCR_CREDS_USR}" "${OCR_CREDS_PSW}"
+
+  if [ $SETUP_HARBOR == true ]; then
+    install_new_helm_version
+    deploy_contour
+    deploy_harbor
+    load_images
+  fi
+
+  docker login ${REGISTRY} -u ${PRIVATE_REGISTRY_USR} -p ${PRIVATE_REGISTRY_PSW}
```
Comment (theabmitra) [inline ci/scripts/prepare_private_registry_jenkins_at_environment.sh:141]: This is being repeated before . Please remove

File: tests/e2e/registry/registry_test.go [hunk 9]
```diff
@@ -17,10 +17,13 @@ import (
 )
 
 const (
-	waitTimeout     = 2 * time.Minute
-	pollingInterval = 10 * time.Second
+	waitTimeout         = 2 * time.Minute
+	pollingInterval     = 10 * time.Second
+	harborEphemeral     = "Harbor ephemeral"
```
Comment (theabmitra) [inline tests/e2e/registry/registry_test.go:22]: This should be only Harbor as we decided to remove other forms of Harbor

File: tests/e2e/registry/registry_test.go [hunk 10]
```diff
@@ -75,14 +78,25 @@ var _ = t.Describe("Private Registry Verification",
...
-								fmt.Sprintf("FAIL: The image for the pod %s in initContainers, doesn't starts with expected registry URL prefix %s, image name %s", pod.Name, registry, pod.Spec.InitContainers[k].Image))
+							assertImageRegistryURLOfPod(pod.Name, pod.Spec.InitContainers[k].Image, imagePrefix, "initContainers")
 						}
 					}
 				}
 			})
 	})
+
+func assertImageRegistryURLOfPod(podName string, image string, imagePrefix string, containersStr string) {
+	if imageRegistrySelection == harborEphemeral {
```
Comment (theabmitra) [inline tests/e2e/registry/registry_test.go:92]: this needs to be harbor and not harborEphemeral to be in sync with other descriptions

File: ci/scripts/prepare_private_registry_jenkins_at_environment.sh (hunk 6 above)
Comment (pshirodkar-orcl) [reply ci/scripts/prepare_private_registry_jenkins_at_environment.sh:62]: sorry which space are you referring to?

File: ci/scripts/prepare_private_registry_jenkins_at_environment.sh (hunk 2 above)
Comment (pshirodkar-orcl) [reply ci/scripts/prepare_private_registry_jenkins_at_environment.sh]: Done

File: ci/scripts/prepare_private_registry_jenkins_at_environment.sh (hunk 3 above)
Comment (pshirodkar-orcl) [reply ci/scripts/prepare_private_registry_jenkins_at_environment.sh]: Done

File: ci/scripts/prepare_private_registry_jenkins_at_environment.sh (hunk 4 above)
Comment (pshirodkar-orcl) [reply ci/scripts/prepare_private_registry_jenkins_at_environment.sh]: Done

File: ci/scripts/prepare_private_registry_jenkins_at_environment.sh (hunk 5 above)
Comment (pshirodkar-orcl) [reply ci/scripts/prepare_private_registry_jenkins_at_environment.sh]: Done

File: ci/scripts/prepare_private_registry_jenkins_at_environment.sh (hunk 7 above)
Comment (pshirodkar-orcl) [reply ci/scripts/prepare_private_registry_jenkins_at_environment.sh]: Done

File: ci/scripts/prepare_private_registry_jenkins_at_environment.sh (hunk 8 above)
Comment (pshirodkar-orcl) [reply ci/scripts/prepare_private_registry_jenkins_at_environment.sh]: Done. Removed.

File: tests/e2e/registry/registry_test.go (hunk 9 above)
Comment (pshirodkar-orcl) [reply tests/e2e/registry/registry_test.go]: Done.

File: Jenkinsfile (hunk 1 above)
Comment (pshirodkar-orcl) [reply Jenkinsfile]: With the jenkins agent templates configuration in place now, I have reverted the agent label in all the affected jenkinsfiles back to their original.

**Review by michael-cico — DISMISSED** [review]

**Review by markxnelson — CHANGES_REQUESTED** [review]
please see comments in slack

**Review by markxnelson — CHANGES_REQUESTED** [review]
address unapproved third party items and resubmit please

**Pull request closed without merging**